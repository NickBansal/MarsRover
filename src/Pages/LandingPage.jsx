import React from "react";
import NasaLogo from "../Stylesheets/Images/NasaLogo.png";
import { Link } from "@reach/router";
import "../Stylesheets/LandingPage.css";

const LandingPage = () => {
  return (
    <div className="LandingPage">
      <Link to="/search">
        <img src={NasaLogo} alt="Nasa logo" />
      </Link>
      <h1>Nasa Image Search Page</h1>
      <p>Please click the picture to start your search</p>
    </div>
  );
};

export default LandingPage;
