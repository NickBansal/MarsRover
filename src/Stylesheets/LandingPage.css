.LandingPage {
    height: 70vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center
}

.LandingPage img {
    opacity: 0;
    height: 300px;
    transform: translateX(-200px);
    animation: 1 slidein 1s forwards
}

.LandingPage h1 {
    font-size: 30px;
    opacity: 0;
    transform: translateX(200px);
    animation: 1 slidein 1s forwards;
    animation-delay: 0.5s
}

.LandingPage p {
    margin: 0;
    opacity: 0;
    animation: 1 slidein 1s forwards;
    animation-delay: 1.2s
}

@keyframes slidein {
    100% {
        opacity: 1;
        transform: translateX(0);
    }
}

@media (max-width: 600px) {
    .LandingPage img {
        height: 250px
    }

    .LandingPage h1 {
        font-size: 20px
    }
}