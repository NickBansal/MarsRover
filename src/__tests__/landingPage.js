import "../setupTest";
import React from "react";
import LandingPage from "../Pages/LandingPage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<LandingPage />", () => {
  it("Should match the page with a snapshot", () => {
    const wrapper = shallow(<LandingPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
