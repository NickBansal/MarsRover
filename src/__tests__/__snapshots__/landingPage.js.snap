// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> Should match the page with a snapshot 1`] = `
<div
  className="LandingPage"
>
  <ForwardRef
    to="/search"
  >
    <img
      alt="Nasa logo"
      src="NasaLogo.png"
    />
  </ForwardRef>
  <h1>
    Nasa Image Search Page
  </h1>
  <p>
    Please click the picture to start your search
  </p>
</div>
`;
